includes/modules/searchandreplace/class-pb-search.php

```diff
@@ -37,7 +49,19 @@ function search_for_pattern( $search, $limit, $offset, $orderby ) {
 			if ( ! ini_get( 'safe_mode' ) ) {
 				set_time_limit( 0 );
 			}
-			return $this->find( '@' . preg_quote( $search, '@' ) . '@', $limit, $offset, $orderby );
+			if ( $this->regex ) {
+				// First test that the search and replace strings are valid regex
+				set_error_handler( array( &$this, 'regex_error' ) );
+				// @codingStandardsIgnoreLine
+				$valid = @preg_match( $search, '', $matches );
```

Comment:
This turned out to be more tricky than I imagined… but I did extensive testing and my code did hold up. (I even found undocumented behaviour of `preg_*` while testing all possible characters at various positions.)